### ISSUE TITLE
Response Rule for RSS

### ISSUE THREAD

**Issue by t2YU2m8l83** [dangkyokhoang/Man-in-the-Middle#10]
Since Firefox 64 removed the rss reader I've tried to implement my own with men in the middle. Sadly it seems `responseHeaders.set('Content-Type', 'text/html; charset=UTF-8');` doesn't change the content type which is necessary to get things working.

**Comment by dangkyokhoang** [comment]
Header Rule
![Man in the Middle - Header Rule - Modify response header](https://user-images.githubusercontent.com/14311741/50020241-f794b580-0007-11e9-9eb0-c31d8f20423c.png)

Try visiting http://rss.nytimes.com/services/xml/rss/nyt/World.xml
![Man in the Middle - Header Rule - Modify response header](https://user-images.githubusercontent.com/14311741/50020463-95888000-0008-11e9-9c1e-550a85285e35.png)
\> no more xml viewer!

Make sure you modify the header using Header Rule, and then `Ctrl + F5` to prevent cache. You may only access but not modify `responseHeaders` in Response Rules.

**Comment by t2YU2m8l83** [comment]
If I try to open a page linking to an .rss file, firefox still tries to download it instead of displaying it as text. The response header is also unaltered in this case. Example: [https://www.heise.de/ct/uplink/ctuplinkvideohd.rss](https://www.heise.de/ct/uplink/ctuplinkvideohd.rss)

**Comment by dangkyokhoang** [comment]
As you can see below;
![Man in the Middle - Modify headers](https://user-images.githubusercontent.com/14311741/50044834-32582580-00bc-11e9-9717-3ed603eb57a4.png)
The Content Type header was modified
![Man in the Middle - Modify headers](https://user-images.githubusercontent.com/14311741/50044885-c0341080-00bc-11e9-810a-2f35e57acea8.png)

Well, there's however a **[bug](https://github.com/dangkyokhoang/Man-in-the-Middle/projects/1#card-15795329)** - the arg `responseHeaders` becomes unavailable when the `Header type` field value is changed so you might need to reload the extension for the rule to work. Go to the Options page, open DevTools > Console, and run `browser.runtime.reload();`.
![Man in the Middle - Modify headers](https://user-images.githubusercontent.com/14311741/50044870-806d2900-00bc-11e9-9bca-a73511d0fa34.png)

**Comment by dangkyokhoang** [comment]
A patch was released. Update the extension and you no longer need to reload the extension to apply new Header Rules.

**Comment by t2YU2m8l83** [comment]
I've done some testing and if you have all three of the following rules set, response rules won't apply to URLs ending with a filename.
`user_pref("extensions.databaseSchema", 27);
user_pref("privacy.resistFingerprinting", true);
user_pref("network.http.referer.spoofSource", true);`

EDIT: Now it magically works in my browser even with all three rules activated...

**Comment by dangkyokhoang** [comment]
Both `Header Rule` and `Response Rule` work for me.
````
about:config
    extensions.databaseSchema;27
    privacy.resistFingerprinting;true
    network.http.referer.spoofSource;true
````
Header Rule
![Man in the Middle - Modify response body](https://user-images.githubusercontent.com/14311741/50054460-9b9b6f80-0174-11e9-9e8b-e7f1efc10351.png)
Response Rule
![Man in the Middle - Modify response body](https://user-images.githubusercontent.com/14311741/50054347-84a84d80-0173-11e9-96bd-0eda343d428b.png)
The result
![Man in the Middle - Modify response body](https://user-images.githubusercontent.com/14311741/50054474-bec61f00-0174-11e9-9489-aecf46ed5224.png)

The rules still work. The pref `resistFingerprinting` is to reduce HighResTime precision and `spoofSource` is to spoof referrer header. They both should not be the cause I think.

You can try `throw`ing the `responseBody`, open the Options page's console and visit the [link](https://www.heise.de/ct/uplink/ctuplinkvideohd.rss) to see if it works:
![Man in the Middle - Modify response body](https://user-images.githubusercontent.com/14311741/50054595-26c93500-0176-11e9-8bb8-b4477adfe042.png)
If it really doesn't work then one more bug is to be left open I think :<

**Comment by t2YU2m8l83** [comment]
As I mentioned it suddenly worked. To test if some config settings cause this, I copied my old about:config entries bit by bit over into a fresh Firefox-Portable and the rule stopped working after all three of config entries were set. If even one of them was missing it worked fine. I can no longer reproduce this behaviour and even my regular browser works just fine with all rules set. The test-browser I used had also the latest version 3.3.1 therfore I really don't know what magic happend that everything works perfekt now.
On another node, I also couldn't reproduce the other bug report with a clean browser. Even with an identical about:config (except the plugin ids) the first rule works just fine despite beeing useless in my normal browser.
